aws/resource_aws_neptune_parameter_group.go

```diff
@@ -0,0 +1,266 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/neptune"
+)
+
+func resourceAwsNeptuneParameterGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNeptuneParameterGroupCreate,
+		Read:   resourceAwsNeptuneParameterGroupRead,
+		Update: resourceAwsNeptuneParameterGroupUpdate,
+		Delete: resourceAwsNeptuneParameterGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				StateFunc: func(val interface{}) string {
+					return strings.ToLower(val.(string))
+				},
+			},
+			"family": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "Managed by Terraform",
+			},
+			"parameter": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"apply_method": &schema.Schema{
```

Comment:
The SDK provides two constants we can use for value validation and to ensure the value is correctly lowercase:

```go
ValidateFunc: validation.StringInSlice([]string{
  neptune.ApplyMethodImmediate,
  neptune.ApplyMethodPendingReboot,
}, false),
```